pkg/pillar/cmd/zedmanager/zedmanager.go

```diff
@@ -742,3 +772,76 @@ func handleGlobalConfigDelete(ctxArg interface{}, key string,
 	*ctx.globalConfig = *types.DefaultConfigItemValueMap()
 	log.Functionf("handleGlobalConfigDelete done for %s", key)
 }
+
+func handleZedAgentStatusCreate(ctxArg interface{}, key string,
+	statusArg interface{}) {
+	handleZedAgentStatusImpl(ctxArg, key, statusArg)
+}
+
+func handleZedAgentStatusModify(ctxArg interface{}, key string,
+	statusArg interface{}, oldStatusArg interface{}) {
+	handleZedAgentStatusImpl(ctxArg, key, statusArg)
+}
+
+func handleZedAgentStatusImpl(ctxArg interface{}, key string,
+	statusArg interface{}) {
+	ctxPtr := ctxArg.(*zedmanagerContext)
+	status := statusArg.(types.ZedAgentStatus)
+	if status.RebootCmd {
+		updateBasedOnProfile(ctxPtr, true)
```

Comment:
Why for RebootCmd?

Reply:
We already have code to handle the halting when we shut down, so I wouldn't add some new code for that.

Reply:
We need to gracefully stop apps despite profiles in case of reboot.

Reply:
Unfortunately it is inside zedagent: https://github.com/lf-edge/eve/blob/158d29ae78ada0ee0298b03305e055e7fd9ee477/pkg/pillar/cmd/zedagent/parseconfig.go#L102